apps/sms/test/marionette/generators/thread.js

```diff
@@ -0,0 +1,54 @@
+function ThreadGenerator() {};
+
+ThreadGenerator.uniqueThreadId = 0;
+ThreadGenerator.uniqueMessageId = 0;
+
+ThreadGenerator.injectDependencies = function(client) {
+  client.contentScript.inject(
+    __dirname + '/../mocks/mock_test_storages.js'
+  );
+  client.contentScript.inject(
+    __dirname + '/../mocks/mock_navigator_moz_mobile_message.js'
+  );
+};
+
+ThreadGenerator.generate = function(numberOfMessages, messageType, participants) {
+  if (typeof numberOfMessages === 'undefined') {
```

Comment:
* Is ```Object.assign``` available in the node env? If so we can simplify this code like:
```
const DEFAULT_PARAMETERS = {
   numberOfMessages: 1,
   messageType: 'sms',
   participants: ['+123']
};

var parameters = Object.assign(DEFAULT_PARAMETERS, inputParameters);
```

* In Messages we're trying to use ```variable === undefined``` instead;

* Please, separate ```if``` blocks with a new line.